tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/PurchasingDocumentServiceTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using FluentAssertions;
+using NHSD.BuyingCatalogue.Ordering.Api.Services.DocumentService;
+using NHSD.BuyingCatalogue.Ordering.Common.UnitTests.Builders;
+using NHSD.BuyingCatalogue.Ordering.Domain;
+using NUnit.Framework;
+
+namespace NHSD.BuyingCatalogue.Ordering.Api.UnitTests.Services
+{
+    [TestFixture]
+    [Parallelizable(ParallelScope.All)]
+    internal sealed class PurchasingDocumentServiceTests
+    {
+        [TestCase(true, false)]
+        [TestCase(false, true)]
+        [TestCase(false, false)]
+        public void CreateDocumentAsync_ArgumentsAreNull_ReturnNullArgumentException(bool hasStream, bool hasOrder)
+        {
+            var purchasingDocumentService = new PurchasingDocumentService();
```

Comment:
I wonder if this test would be easier to read if the creation of the `PurchasingDocumentService` object could be moved into the Assert.Throws function. Also the stream and order objects could use one line as well e.g. 

`hasOrder ? OrderBuilder.Create().Build() : null`